modules/tftp/tftp_api.rb

```diff
@@ -34,6 +34,10 @@ def create_default(variant)
       end
     end
 
+    post "/fetch_boot_image" do
+      log_halt(400, "TFTP: Failed to fetch boot file: ") { Proxy::TFTP.fetch_boot_image(params[:path], params[:url]) }
```

Comment:
What is the reason to present this as a HTTP 400 Bad Request? I see it was presented in the old code too, but I'm not sure it makes sense. HTTP 400 errors means the client did something wrong but isn't it really on the server side that there's a problem if it fails? So HTTP 5xx?

Reply:
I guess this could be discussed, since it might as well be that an incorrect URL has been specified, which would then mean 404, can't find what you are looking for.

This would probably need a more sophisticated differentiation on what error actually occurred.

Reply:
Change to 500 http error. Add another check on input parameters which raises http error 400.